File: eio2024-lv/2024-10-07-lv/captcha/solution/sol2.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
	vector<string> pildid(5);
	for (auto & pilt : pildid) {
		cin >> pilt;
	}
	int n;
	cin >> n;
	vector<string> klikid;
	for (int i = 0; i < n; ++i) {
		string pilt;
		cin >> pilt;
		const auto it = find(klikid.begin(), klikid.end(), pilt);
		if (it == klikid.end()) {
			klikid.push_back(pilt);
		} else {
			klikid.erase(it);
		}
	}

	vector<string> logi;
	for (int i = 0; i < pildid.size(); ++i) {
		while (klikid.size() > i and klikid[i] != pildid[i]) {
			logi.push_back(klikid[i]);
			klikid.erase(klikid.begin() + i);
		}
		if (klikid.size() <= i) {
			logi.push_back(pildid[i]);
		}
	}

	cout << logi.size() << '\n';
	for (const auto & pilt : logi) {
		cout << pilt << '\n';
	}
}

File: eio2024-lv/2024-10-07-lv/captcha/solution/sol2.py
pildid = []
for _ in range(5):
    pildid.append(input())

klikid = []
n = int(input())
for _ in range(n):
    pilt = input()
    if pilt in klikid:
        klikid.remove(pilt)
    else:
        klikid.append(pilt)

logi = []
for i in range(len(pildid)):
    while len(klikid) > i and klikid[i] != pildid[i]:
        logi.append(klikid[i])
        klikid.pop(i)
    if len(klikid) <= i:
        logi.append(pildid[i])

print(len(logi))
for pilt in logi:
    print(pilt)
